docs/app/pages/components/sections/panels/item-display-panel/item-display-panel.component.ts

```diff
@@ -0,0 +1,177 @@
+    previousEnabled: boolean = true;
+    nextEnabled: boolean = true;
+    
+    sparkBarColor: string;
+    sparkTrackColor: string;
+
+    htmlCode = require('./snippets/app.html');
+    tsCode = require('./snippets/app.ts');
+    modalDoc = require('./modalDOC.html');
+    modalPdf = require('./modalPDF.html');
+    modalPpt = require('./modalPPT.html');
+
+    items: Item[] = [{
+        id: 1,
+        name: chance.name(),
+        dateString: '3 Oct 2015',
+        document: 'Document 4.ppt',
+        extension: '.ppt',
+        storage: '95.25',
+        active: false,
+        panel: {
+            title: 'Site Detail - UX Aspects (PPT)',
+            main: this.modalPpt,
+            top: 53
+        }
+        }, {
+            id: 2,
+            name: chance.name(),
+            dateString: '3 Oct 2015',
+            document: 'Document 9.pdf',
+            extension: '.pdf',
+            storage: '15.25',
+            active: true,
+            panel: {
+                title: 'Site Detail - UX Aspects (PDF)',
+                main: this.modalPdf,
+                top: 53
+            }
+        }, {
+            id: 3,
+            name: chance.name(),
+            dateString: '3 Oct 2015',
+            document: 'Document 14.doc',
+            extension: '.doc',
+            storage: '25.25',
+            active: false,
+            panel: {
+                title: 'Site Detail - UX Aspects (DOC)',
+                main: this.modalDoc,
+                top: 53
+            }
+        }, {
+            id: 4,
+            name: chance.name(),
+            dateString: '3 Oct 2015',
+            document: 'Document 29.pdf',
+            extension: '.pdf',
+            storage: '15.25',
+            active: true,
+            panel: {
+                title: 'Site Detail - UX Aspects (PDF)',
+                main: this.modalPdf,
+                top: 53
+            }
+        }, {
+            id: 5,
+            name: chance.name(),
+            dateString: '3 Oct 2015',
+            document: 'Document 34.doc',
+            extension: '.doc',
+            storage: '15.25',
+            active: false,
+            panel: {
+                title: 'Site Detail - UX Aspects (DOC)',
+                main: this.modalDoc,
+                top: 53
+            }
+        }
+    ];
+
+    constructor(colorService: ColorService) {
+        this.sparkTrackColor = colorService.getColor('accent').setAlpha(0.2).toRgba();
+        this.sparkBarColor = colorService.getColor('accent').toHex();
+    }
+
+    show(panel: ItemDisplayPanelComponent, $event: MouseEvent, id: number) {
+        $event.stopPropagation();
+        this.selected = id;
+        this.updatePanel();
+        panel.show();
+    }
+
+    previous() {
+        this.selected--;
+        this.updatePanel();
+    }
+
+    next() {
+        this.selected++;
+        this.updatePanel();
+    }
+
+    updatePanel() {
+        this.title = this.items[this.selected - 1].panel.title;
```

Comment:
We should be able to get rid of all these variables, just have one variable called `selectedItem` or something, and in your html just do `selectedItem?.id` for example.